cloud_dashboard/src/service/utility.ts

```diff
@@ -206,3 +221,69 @@ export const getEntityDataAll = async (entityTypeId: string) => {
   }
   return output;
 };
+
+/**
+ * Getter of LaunchTemplateView's URL for CloudContext.
+ * @param cloudContext CloudContext
+ * @returns URL
+ */
+export const getLaunchTemplateViewUrl = (cloudContext: CloudContext) => {
+  if (cloudContext.name === 'ALL') {
+    return `/${cloudContext.cloudServiceProvider as string}/server_template`;
+  } else {
+    return `/server_template/${cloudContext.cloudServiceProvider as string}/${cloudContext.name}`;
+  }
+};
+
+/**
+ * Read entity's data for convert entity's data by JSON:API.
+ * @param entityColumnList Information about the entities that will be loaded in advance.
+ * @returns Data cache.
+ */
+export const readDataCache = async (entityColumnList: EntityColumn[]) => {
+  const dataCache: { [key: string]: EntityData[] } = {};
+  for (const entityColumn of entityColumnList) {
+    if (entityColumn.type !== 'join') {
+      continue;
+    }
+    const entityTypeId = entityColumn.info.entityTypeId;
+    if (entityTypeId in dataCache) {
+      continue;
+    }
+    dataCache[entityTypeId] = await getEntityDataAll(entityTypeId);
+  }
+  return dataCache;
+};
+
+/**
+ * Convert EntityData to DataRecord.
+ * 
+ * @param rawDataList List of EntityData.
+ * @param entityColumnList List of EntiyColumn.
+ * @param cloudContext cloud_context.
+ * @param dataCache Data cache of EntityData.
+ * @returns List of DataRecord.
+ */
+export const convertEntityData = (
+  rawDataList: EntityData[],
+  entityColumnList: EntityColumn[],
+  cloudContext: CloudContext,
+  dataCache: { [key: string]: EntityData[] }) => {
+  const newDataRecordList: DataRecord[] = [];
+  for (const rawRecord of rawDataList) {
+    const dataRecord: DataRecord = {
+      id: rawRecord.id,
+      value: {}
+    };
+    for (const launchTemplateColumn of entityColumnList) {
+      const rawValue = rawRecord.attributes[launchTemplateColumn.name];
+      if (launchTemplateColumn.name === 'cloud_context') {
+        dataRecord.value[launchTemplateColumn.name] = cloudContext.labelName;
+      } else {
+        dataRecord.value[launchTemplateColumn.name] = convertDataForUI(rawValue, launchTemplateColumn, dataCache);
+      }
```

Comment:
@RyoYamashita765 Can we refactor:

```suggestion
      dataRecord.value[launchTemplateColumn.name] = launchTemplateColumn.name === 'cloud_context'
        ? cloudContext.labelName
        :  convertDataForUI(rawValue, launchTemplateColumn, dataCache);
```